instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/config/internal/InstrumentationConfig.java

```diff
@@ -107,4 +108,18 @@ default List<String> getList(String name) {
    * {@code key=value,anotherKey=anotherValue}. The returned map is unmodifiable.
    */
   Map<String, String> getMap(String name, Map<String, String> defaultValue);
+
+  /**
+   * Returns a {@link DeclarativeConfigProperties} for the given instrumentation name, or {@code
+   * null} if no declarative configuration is available for that instrumentation.
+   *
+   * <p>Declarative configuration is used to configure instrumentation properties in a declarative
+   * way, such as through YAML or JSON files.
+   *
+   * @param instrumentationName the name of the instrumentation
+   * @return the declarative configuration properties for the given instrumentation name, or {@code
+   *     null} if not available
+   */
+  @Nullable
+  DeclarativeConfigProperties getDeclarativeConfig(String instrumentationName);
```

Comment:
may want to rename this param name in the future since we've discussed it also being "vendor" node, or "common"